plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreExecutionContext.java

```diff
@@ -111,6 +111,10 @@ void setDefaultCatalog(@NotNull DBRProgressMonitor monitor, @NotNull PostgreData
                     disconnect();
                     setOwnerInstance(catalog);
                     connect(monitor, null, null, null, false);
+                    // Catalog has been changed. Get the new one and propagate isolated flag
```

Comment:
Don't understand this logic. We already have new instance (it is catalog). Why read it again why set isolated flag?

Reply:
It's the same situation as here https://github.com/dbeaver/dbeaver/pull/36404/files#diff-39b703e635479b90005852d1883628ef40d44433da8992522bd6bea810ea6ffbR125-R128
After disconnecting and connecting, the catalog is a different object, and it loses the isolatestContext property value.